plugins/newsdownloader.koplugin/feed_view.lua

```diff
@@ -0,0 +1,198 @@
+local logger = require("logger")
+local _ = require("gettext")
+
+local FeedView = {
+    URL = "url",
+    LIMIT = "limit",
+    DOWNLOAD_FULL_ARTICLE = "download_full_article",
+    INCLUDE_IMAGES = "include_images",
+    ENABLE_FILTER = "enable_filter",
+    FILTER_ELEMENT = "filter_element"
+}
+
+function FeedView:getList(feed_config, callback, edit_feed_attribute_callback, delete_feed_callback)
+    local view_content = {}
+    -- Loop through the feed.
+    for idx, feed in ipairs(feed_config) do
+        local feed_item_content = {}
+        local feed_list_content = {}
+
+        local vc_feed_item = FeedView:getItem(
+            idx,
+            feed,
+            edit_feed_attribute_callback,
+            delete_feed_callback
+        )
+
+        if not vc_feed_item then
+            logger.warn('NewsDownloader: invalid feed config entry', feed)
+        else
+            feed_item_content = FeedView:flattenArray(feed_item_content, vc_feed_item)
+            local url = feed[1]
+            table.insert(
+                view_content,
+                {
+                    url,
+                    "",
+                    callback = function()
+                        -- Here is where we trigger the single feed item display
+                        callback(feed_item_content)
+                    end
+                }
+            )
+            -- Insert a divider.
+            table.insert(
+                view_content,
+                "---"
+            )
+        end
+    end
+    return view_content
+end
+
+function FeedView:getItem(id, feed, edit_feed_callback, delete_feed_callback)
+
+    logger.dbg("NewsDownloader:", feed)
+
+    local url = feed[1]
+    local limit = feed.limit
+
+    -- If there's no URL or limit we don't care about this
+    -- because we can't use it.
+    if not url and limit then
+        return nil
+    end
+
+    -- Collect this stuff for later, with the single view.
+    local download_full_article = feed.download_full_article == nil
+        or feed.download_full_article
+    local include_images = not never_download_images
+        and feed.include_images
+    local enable_filter = feed.enable_filter
+        or feed.enable_filter == nil
+    local filter_element = feed.filter_element
+        or feed.filter_element == nil
+
+    -- TODO: Strip the http:// or https:// from the URL
+    local sPre, sLink, sPost = url:match( "(.+)%s+(https?%S+)%s+(.*)$" )
+    local pretty_url = sLink
+
+    local view_content = {}
+    local vc = {
+        {
+            "URL",
+            url,
+            callback = function()
+                edit_feed_callback(
+                    id,
+                    FeedView.URL,
+                    url
+                )
+            end
+        },
+        {
+            "Limit",
```

Comment:
As @poire-z already said but this way it might be more visible for potential onlookers. Otherwise it won't be translated.
```suggestion
            _("Limit"),
```

Reply:
Thanks for the reminder! I keep forgetting to wrap the strings. 🙃

Reply:
@roygbyte Oh, but the strings aren't wrapped in _() yet.